example/parquet_reader.cc

```diff
@@ -41,254 +39,229 @@ static string ByteArrayToString(const ByteArray& a) {
   return string(reinterpret_cast<const char*>(a.ptr), a.len);
 }
 
-void* read_parquet(char* filename);
-
-// Simple example which prints out the content of the Parquet file
-int main(int argc, char** argv) {
-
-  if (argc < 2) {
-    cerr << "Usage: parquet_reader <file>" << endl;
-    return -1;
-  }
-
-  void *column_ptr = read_parquet(argv[1]);
-
-  // an example to use the returned column_ptr
-  // printf("%-"COL_WIDTH"d\n",((int32_t *)(((int32_t **)column_ptr)[0]))[0]);
-
+int ByteCompare(const ByteArray& x1, const ByteArray& x2) {
+  int len = ::min(x1.len, x2.len);
+  int cmp = memcmp(x1.ptr, x2.ptr, len);
+  if (cmp != 0) return cmp;
+  if (len < x1.len) return 1;
+  if (len < x2.len) return -1;
   return 0;
 }
 
+string type2String(Type::type t) {
+  switch(t) {
+    case Type::BOOLEAN:
+      return "BOOLEAN";
+      break;
+    case Type::INT32:
+      return "INT32";
+      break;
```

Comment:
You are right, semantically these breaks are redundant. However, some compilers may complain about missing breaks in a switch statement, so it's safer to keep them.